File: scripts/test_solutions/155_min_stack.cpp
class MinStack {
private:
    vector<int> stack;
    vector<int> min_stack;

public:
    MinStack() {
    }

    void push(int val) {
        stack.push_back(val);
        int min_val = min_stack.empty() ? val : min(val, min_stack.back());
        min_stack.push_back(min_val);
    }

    void pop() {
        if (!stack.empty()) {
            stack.pop_back();
            min_stack.pop_back();
        }
    }

    int top() {
        if (stack.empty()) return INT_MAX;  // sentinel value for empty stack
        return stack.back();
    }

    int getMin() {
        if (min_stack.empty()) return INT_MAX;  // sentinel value for empty stack
        return min_stack.back();
    }
};

File: scripts/test_solutions/155_min_stack.js
var MinStack = function() {
    this.stack = [];
    this.minStack = [];
};
MinStack.prototype.push = function(val) {
    this.stack.push(val);
    const min = this.minStack.length ? Math.min(val, this.minStack[this.minStack.length - 1]) : val;
    this.minStack.push(min);
};
MinStack.prototype.pop = function() { this.stack.pop(); this.minStack.pop(); };
MinStack.prototype.top = function() { return this.stack[this.stack.length - 1]; };
MinStack.prototype.getMin = function() { return this.minStack[this.minStack.length - 1]; };
